Kitodo-API/src/main/java/org/kitodo/api/filemanagement/FileManagementInterface.java

```diff
@@ -38,24 +40,89 @@
      */
     URI create(URI parentFolderUri, String name, boolean file) throws IOException;
 
+    /**
+     * Attempts to get locks on one or more files. There are only two results:
+     * either, all locks can be granted, or none of the requested locks are
+     * granted. In the former case, the conflicts map in the locking result is
+     * empty, in the latter case it contains for each conflicting file the users
+     * who hold a conflicting lock on the file. If no locks have been granted,
+     * the call to {@code close()} on the locking result is meaningless, meaning
+     * that leaving the try-with-resources statement will not throw an
+     * exception. Just to mention that.
+     * 
+     * @param user
+     *            A human-readable string that identifies the user or process
+     *            requesting the locks. This string will later be returned to
+     *            other users if they try to request a conflicting lock.
+     * @param requests
+     *            the locks to request
+     * @return An object that manages allocated locks or provides information
+     *         about conflict originators in case of error.
+     * @throws IOException
+     *             if the file does not exist or if an error occurs in disk
+     *             access, e.g. because the write permission for the directory
+     *             is missing
+     */
+    LockingResult tryLock(String user, Map<URI, LockingMode> requests) throws IOException;
+
     /**
      * Opens an OutputStream to a given uri.
      *
      * @param uri
      *            the uri to write to
      * @return an writable OutputStream
+     * @deprecated This write function creates an exclusive lock with the
+     *             meaningless user name “System”. Therefore, this writing
+     *             function should not be used anymore. Use
+     *             {@link #write(URI, LockingResult)} instead.
      */
+    @Deprecated
```

Comment:
The thing is, since the interface does not get passed a user, only one standard user “System” is ever used for the locks, which is stupid. And that every read requires an exclusive lock, I do not think so either. We can discuss that, but with whom?

Reply:
The thing is, since the interface does not get passed a user, only one standard user “System” is ever used for the locks, which is stupid. And that every read requires an exclusive lock, I do not think so either. We can discuss that, but with whom?

Reply:
The thing is, since the interface does not get passed a user, only one standard user “System” is ever used for the locks, which is stupid. And that every read requires an exclusive lock, I do not think so either. We can discuss that, but with whom?